building/tracks/syllabus/README.md

```diff
@@ -0,0 +1,165 @@
+We simplify and provide code stubs where possible.
+This reduces the cognitive burden of getting started and provides the time and space for the knowledge to sink in.
+By taking this approach we're not saying that the student doesn't need to know these things, we're saying that they don't need to know them _yet_.
+
+## Basic structure
+
+Exercises are structured as a tree with an introductory exercise at the top as the starting point.
+Later exercises teach concepts that depend on having understood concepts that are taught earlier.
+
+## Porting and borrowing
+
+It can be worth looking at how other language tracks have built out their concept exercises.
+You can find examples of Concept Exercises from other language tracks [here](https://exercism.org/docs/building/tracks/stories).
+
+That said, if you decide to use other exercises as a starting point for your own, be careful to ensure that the resulting exercise is about the concept as it exists in your language.
+Sometimes concepts differ subtly, sometimes radically.
+Sometimes concepts don't exist at all.
+
+The syllabus, and therefore the concept tree, should represent the concepts that exist in the language.
+
+Don't include concepts just because other tracks do.
+
+In some cases it might be tempting to put a concept in because people often have to work around it by using concepts that do exist.
+Rather than doing this, introduce the concept that the language _does_ use, and consider adding an exercise that explains how to use it in that type of situation.
+
+For example, in Go there are no enums.
+Instead the Go concept tree introduces constants, and teaches how to use constants in the type of situation where you might use enums in other languages.
+
+## Asking for help
+Don't hesitate to ask for help.
+It's better to ask up front or while working on an exercise rather than discussing during code review.
+
+## Getting started
+
+Our experience has taught us that the most pragmatic way to develop a syllabus is to grow the concept tree organically, starting with the simplest concepts.
+We don't have to get everything right up front, but it's really useful to not think too far ahead.
+
+We start with concepts that are easy.
+
+Some things are easy because they are inherently simple.
+Other things are not all that simple, but feel easy because they are familiar.
+Familiar is good.
+Familiar is not confusing.
+
+While the endpoint is to write idiomatic code, the stepping stones to get there are not always idiomatic.
+Using what is familiar—even if it is not a great example of code in that language—helps move a student more quickly toward the goal of code that is more typical of the language.
+
+### Developing the first exercise
+
+Rather than trying to map out the entire concept tree up front, just start with the first exercise.
+The goal of the first exercise is to show the student what a small piece of code looks like.
+Most of the code will be provided as a stub.
```

Comment:
This is referring to the `basics` exercises, right? As those exercises usually aim to have student write functions or methods from scratch, often there is one a very minimal stub.

Reply:
Ah, I thought tracks mostly provided stubs for the functions and then had exceptions or comments for people to fill them in. Let me reword.

Reply:
That is true for all exercises but the first one.

Reply:
My understanding was what @kytrinyx described, that most code is already in the stub. This is how it is in JS and Go (see links below). Imo it is ok to have the stubs in basics. Especially for statically typed + compiled languages, the exercise would be rather difficult (and therefore potentially discouraging) otherwise. To write the function signatures in lasagna in a language like Go, you would need to know the type names for the number types which are usually introduced in an exercise that comes after basics that introduces numbers. Also, in a compiled+typed language, no test passes until you wrote ALL the functions signatures with some valid function body that has the correct return type.

https://github.com/exercism/go/blob/main/exercises/concept/lasagna/lasagna.go
https://github.com/exercism/javascript/blob/main/exercises/concept/lasagna/lasagna.js

The flow in Go and JavaScript is that stubs are provided in all the concept exercises at the beginning. Then, there is an exercise that explicitly teaches "functions" and there the student needs to write everything after reading the proper introduction for it.

I always regarded "basics" as the "hello-world" of concept exercises. Since this is the learning mode, you get the additional documentation. For me, the learning objective of the basics exercise wasn't really "understanding functions" but rather "learn the minimum you need to know to be able to actually start with the learning journey of the track, i.e. solve exercises like annlyns infiltration (booleans) etc" and "be able to read/understand the stub code".

I think for interpreted + dynamically typed languages, no stubs are probably ok but I wouldn't like to see "no stub" as requirement for basics in the official docs.

Reply:
I think I'll hedge here and instead of giving direct guidelines (stubs or no stubs) focus on the goal: being able to start learning with the least about of friction.

Reply:
👍